manifests/journal_upload.pp

```diff
@@ -0,0 +1,37 @@
+# @api private
+# @summary This class manages and configures journal-upload.
+# @see https://www.freedesktop.org/software/systemd/man/journald.conf.html
+#
+# @param package_name
+#  name of the package to install for the functionality
+#
+class systemd::journal_upload (
+  Optional[String[1]] $package_name = undef,
+) {
+  assert_private()
+
+  if $package_name {
+    stdlib::ensure_packages($package_name)
+  }
+
+  service { 'systemd-journal-upload':
+    ensure => running,
```

Comment:
to ensure we don't have a corrective change after a reboot, should we add `enable => true`?